documentation/processing.tex

```diff
@@ -1 +1,315 @@
+
+\emph{Copy} and \emph{Macro} processing kinds state that manager will process source code that contains copy or macro definition respectively. Hence, \emph{only} copy definition processor or macro definition processor is initialized. Also, all but macro statement provider are initialized. That is because macro definitions would pollute statement sources (i.e. when parse library provider is called within macro call). Also, no macros will be visited nor needed as a statement source when processing new source code. This enumeration is passed when analyzer has reference semantics.
+
+
+\subsection{Statement field parser}
+\label{lab06:field_parser}
+
+Statement field parser is an interface passed to the statement providers by processing manager implemented by the parser (see \cref{lab06:parser}).
+
+Firstly, it is used during statement passing. In some cases provider is requested a concrete format of a string-stored statement. The string is re-parsed with the according format. Then the field is returned back to the statement provider. 
+
+Another use of field parser is in opencode processor as model statements are resolved there. After variable symbol substitution, the resulting string field is re-parsed with field parser.
+
+\subsection{Instruction processors}
+\label{lab06:instr_proc}
+
+Used by ordinary processor, instruction processors are helper classes that divide processing of HLASM instruction types.
+
+As format of some instruction kinds can be rather complicated, instruction processors contain \emph{Instruction format validators} (see \cref{checker}). They check the statement to validate correctness of used operand format.
+
+During the work of the processors, they encounter \emph{expression} structures (see \cref{ca_expr:logic}, \cref{mach_expr}). Expressions need to be evaluated for processing to be correctly finished.
+
+There are four specialized instruction processors:
+\paragraph*{Macro IP} looks up for macro definition in HLASM context tables and calls it.
+\paragraph*{Assembler and Machine IP} processes assembler and machine instructions (see \cref{asm_instrs} \cref{mach_instr}) to retain consistency in HLASM context tables.
+
+\paragraph*{Conditional assembly IP} executes conditional assembly instructions (see \cref{ca_instr}). 
+
+\vspace{5mm}
+
+See the current list of processed instruction in \cref{tab06:instr_proc}.
+
+\begin{table}
+	\centering
+	\begin{tabular}{lr}
+		\textbf{IP}                   &                  \textbf{Processed instructions} \\ \toprule
+		\textbf{Assembler}            & *SECT, COM, LOCTR, EQU, DC, DS, COPY, EXTRN, ORG \\
+		\textbf{Machine}              &                                      \emph{NONE} \\
```

Comment:
zmenil by som na "checked only"... a expressiony su aj v machine processovane predsa

Reply:
"... a expressiony su aj v machine processovane predsa"

aku zmenu si tym myslel?

Reply:
ani neviem.. proste ze v sirokom vyzname slova mi pride ze aj machine expressny su processovane, moze to niekoho zmiast. Ale chapem ze v celej kapitole sa asi 'processing' pouziva ako interpretation